src/main/java/com/google/api/codegen/config/SampleContext.java

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import com.google.api.codegen.metacode.InitCodeContext.InitCodeOutputType;
+import com.google.api.codegen.viewmodel.CallingForm;
+import com.google.auto.value.AutoValue;
+
+/** The context of transforming a sample into a view model. */
+@AutoValue
+public abstract class SampleContext {
+
+  /**
+   * The unique sample ID generated by the generator.
+   *
+   * <p>If the user provided a sample ID, the user provided ID will be used as a base ID.
```

Comment:
I don't think this additional description of _how_ we create a unique ID belongs here but rather in the comments to `SampleFileRegistry`, which is the class that implements the algorithm you describe. The comments here should just say it's unique and refer folks to the other class for details. The reasoning is that this way we keep the description close to the implementation, and can make sure they're accurate whenever we tweak anything.

Reply:
SGTM